Arcane, BM

syrifgit opened the pull request:
### Arcane
Arcane Missile final solution (please)

### Beast Mastery
- Simplify barb/thrill/frenzy duration management
- Implement prediction for CoTW barb and KC cooldown refund ticks
- add skeleton for a mend pet spec option for solo play / world content / delves, to be finished at a later date

syrifgit committed:
Arcane, BM

- Arcane Missile final solution (please)
- BM polishing
 - Simplify barb/thrill/frenzy duration management
 - Implement prediction for CoTW barb and KC cooldown refund ticks
 - add skeleton for a mend pet spec option for solo play / world content / delves

syrifgit committed:
Merge branch 'Hekili:thewarwithin' into thewarwithin

Hekili commented on TheWarWithin/MageArcane.lua:
```diff
@@ -2211,4 +2211,4 @@ spec:RegisterSetting( "cancel_pom", false, {
 
-spec:RegisterPack( "Arcane", 20240910.1, [[Hekili:TZ1EtUTns(plUCvYIjYJ1JzCSZknv51xUBTRlztvYBU)4QvuuKuJ4o8HcFmJNRMIF2VUbabbaba58YXjBQkvSTiiqJ(5VUrdUz2MpTzDGxz4MFA(05No9TZMEYSxV4nZoBZ6YBogUz9rp)l9Ua(lPEjW))D5(EPKF(M4mVa81lYQY9HF6qz5XIV)vV6IOYdv7oXpl5vfrjvXELrzP(5E7lX)T)R2SExvuC5hs3St3A)6PVbMZJH(B(PxphM1OGGq6qdl83Sgh6lN(2xoB63xV9txh6Dz92SJHPH51FS(JnpD6Sj1BX)Cb7pFdm61rjVVEB1rCnBh8BE583A8HF3lNJV5NoewV ...
\ No newline at end of file
+spec:RegisterPack( "Arcane", 20240910.1, [[Hekili:T31EpUnos(plbbWXAgNo(vNhZA3azZn3Tj4MmdGZo7FCyTSSKCBTTSKJE0D6dn0N9Tkskksksk1VYKz2byXMelkYI1ZFvXIAwpz9NwVkWRiC9hNoE68XVzYKtMmF8RMoD9QIRpgUE1rp)l8oh(ljEhG))3M57Lq(5RJt9cWxppTmZh(P9ffhZ)Hx8IZJk2xU9e)0dVip6qzSxruAIFM3Uc8F7)I1R2wgfx8(K1B1T2VCYmyopg6V(JVeOI9rbbH0HgM7VEfo0Np(npFY4FOAZNUk07IQnPhdtcZQ(q1hQF64jJQ2G)5m2F(Ay0RIo8UQnLhX1SzWV(5tFJXh(QNpfFZpTpSAZ ...
```
If you updated this export string, you have to increase the version number to current date.

syrifgit replied:
Wouldn't be my pr if I didn't forget something, made new commit

syrifgit committed:
pack string

Hekili commented on TheWarWithin/Priorities/MageArcane.simc:
```diff
@@ -48,7 +48,7 @@ actions.cd_opener+=/arcane_blast,if=buff.presence_of_mind.up
 actions.cd_opener+=/arcane_orb,if=talent.high_voltage&buff.arcane_charge.stack<4,line_cd=10
 actions.cd_opener+=/evocation,if=cooldown.arcane_surge.remains<(gcd.max*3)&cooldown.touch_of_the_magi.remains<(gcd.max*5)
-actions.cd_opener+=/arcane_missiles,if=variable.opener&talent.nether_precision&buff.nether_precision.down,interrupt_if=tick_time>gcd.remains&buff.aether_attunement.down,interrupt_immediate=1,interrupt_global=1,chain=1,line_cd=30
+actions.cd_opener+=/arcane_missiles,if=prev_gcd.1.evocation|prev_gcd.1.arcane_orb,interrupt_if=tick_time>gcd.remains&buff.aether_attunement.down,interrupt_immediate=1,interrupt_global=1,chain=1,line_cd=30
```
This is still not really going to work with the `line_cd` in place -- I recommend removing it since this will only fire if no Arcane Missiles have been cast in the past 30 seconds.  You probably need the `variable.opener` check so it doesn't fire at other times.

syrifgit replied:
Agree on line_cd, I think I had intended to remove it.

As for variable.opener on the missile, that is a condition requirement to even enter the `cd_opener` sublist in the first place, so it would be redundant. (this is a difference from the sim APL that I had to make in order to replicate it in-game)

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -2412,9 +2436,22 @@ spec:RegisterSetting( "barbed_shot_grace_period", 0.5, {
+    width = "normal"
 } )
 
+--[[
```
```
spec:RegisterStateExpr( "pet_healing", function ()
    return settings.pet_healing or 0
end )
```

`actions+=/mend_pet,if=pet.alive&pet.health_pct<pet_healing`

syrifgit committed:
remove line_cd from arcane missile

It's not needed anymore anyways, the prev.gcd conditions completely control the behaviour and make it match simc
